ring/lifecycler_test.go

```diff
@@ -160,6 +160,157 @@ func TestLifecycler_HealthyInstancesCount(t *testing.T) {
 	})
 }
 
+func TestLifecycler_HealthyInstancesInZoneCount(t *testing.T) {
+	ringStore, closer := consul.NewInMemoryClient(GetCodec(), log.NewNopLogger(), nil)
+	t.Cleanup(func() { assert.NoError(t, closer.Close()) })
+
+	var ringConfig Config
+	flagext.DefaultValues(&ringConfig)
+	ringConfig.KVStore.Mock = ringStore
+
+	ctx := context.Background()
+
+	// Add the first ingester to the ring
+	lifecyclerConfig1 := testLifecyclerConfig(ringConfig, "ing1")
+	lifecyclerConfig1.HeartbeatPeriod = 100 * time.Millisecond
+	lifecyclerConfig1.JoinAfter = 100 * time.Millisecond
+	lifecyclerConfig1.Zone = "zone-a"
+
+	lifecycler1, err := NewLifecycler(lifecyclerConfig1, &nopFlushTransferer{}, "ingester", ringKey, true, log.NewNopLogger(), nil)
+	require.NoError(t, err)
+	assert.Equal(t, 0, lifecycler1.HealthyInstancesInZoneCount())
+
+	require.NoError(t, services.StartAndAwaitRunning(ctx, lifecycler1))
+	defer services.StopAndAwaitTerminated(ctx, lifecycler1) // nolint:errcheck
+
+	// Assert the first ingester joined the ring
+	test.Poll(t, 1000*time.Millisecond, true, func() interface{} {
+		return lifecycler1.HealthyInstancesInZoneCount() == 1
+	})
+
+	// Add the second ingester to the ring in the same zone
+	lifecyclerConfig2 := testLifecyclerConfig(ringConfig, "ing2")
+	lifecyclerConfig2.HeartbeatPeriod = 100 * time.Millisecond
+	lifecyclerConfig2.JoinAfter = 100 * time.Millisecond
+	lifecyclerConfig2.Zone = "zone-a"
+
+	lifecycler2, err := NewLifecycler(lifecyclerConfig2, &nopFlushTransferer{}, "ingester", ringKey, true, log.NewNopLogger(), nil)
+	require.NoError(t, err)
+	assert.Equal(t, 0, lifecycler2.HealthyInstancesInZoneCount())
+
+	require.NoError(t, services.StartAndAwaitRunning(ctx, lifecycler2))
+	defer services.StopAndAwaitTerminated(ctx, lifecycler2) // nolint:errcheck
+
+	// Assert the second ingester joined the ring
+	test.Poll(t, 1000*time.Millisecond, true, func() interface{} {
```

Comment:
```suggestion
	test.Poll(t, time.Second, true, func() interface{} {
```

(applies in entire test)